src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2390,6 +2396,346 @@ def remove_triggered_webjob(cmd, resource_group_name, name, webjob_name, slot=No
+        i = i + 1
+
+    if slot is None:
+        swift_connection_info = client.web_apps.get_swift_virtual_network_connection(resource_group_name, name)
+    else:
+        swift_connection_info = client.web_apps.get_swift_virtual_network_connection_slot(resource_group_name,
+                                                                                          name, slot)
+
+    # check to see if the connection would be supported
+    if swift_connection_info.swift_supported is not True:
+        return logger.warning("""Your app must be in an Azure App Service deployment that is
+              capable of scaling up to Premium v2\nLearn more:
+              https://go.microsoft.com/fwlink/?linkid=2060115&clcid=0x409""")
+
+    subnetObj = vnet_client.subnets.get(vnet_resource_group, vnet, subnet)
+    delegations = subnetObj.delegations
+    delegated = False
+    for d in delegations:
+        if d.service_name == "Microsoft.Web/serverFarms":
+            delegated = True
+
+    if not delegated:
+        vnet_client.subnets.create_or_update(vnet_resource_group, vnet, subnet,
+                                             subnet_parameters=Subnet(name="subnet",
+                                                                      address_prefix=subnetObj.address_prefix,
+                                                                      delegations=[Delegation(name="delegation",
+                                                                                              service_name="Microsoft" +
+                                                                                              ".Web/serverFarms")]))
+
+    id_subnet = vnet_client.subnets.get(vnet_resource_group, vnet, subnet)
+    subnet_resource_id = id_subnet.id
+    swiftVnet = SwiftVirtualNetwork(subnet_resource_id=subnet_resource_id,
+                                    swift_supported=True)
+
+    if slot is None:
+        return_vnet = client.web_apps.create_or_update_swift_virtual_network_connection(resource_group_name, name,
+                                                                                        swiftVnet)
+    else:
+        return_vnet = client.web_apps.create_or_update_swift_virtual_network_connection_slot(resource_group_name, name,
+                                                                                             swiftVnet, slot)
+    # reformats the vnet entry, removing unecessary information
+    id_strings = return_vnet.id.split('/')
+    resourceGroup = id_strings[4]
+    mod_vnet = {
+        "id": return_vnet.id,
+        "location": return_vnet.additional_properties["location"],
+        "name": return_vnet.name,
+        "resourceGroup": resourceGroup,
+        "subnetResourceId": return_vnet.subnet_resource_id
+    }
+
+    return mod_vnet
+
+
+def remove_vnet_integration(cmd, name, resource_group_name, slot=None):
+    client = web_client_factory(cmd.cli_ctx)
+    if slot is None:
```

Comment:
you should try to leverage `_generic_site_operation` like the rest code

Reply:
@yugangw-msft , given this means that we need to re-record the tests again, is it ok if this something I take care of in another PR, & let Sofi move on to her other work?